Mathlib/NumberTheory/ModularForms/EisensteinSeries/Finset_Decomposition.lean

```diff
@@ -0,0 +1,108 @@
+/-
+Copyright (c) 2024 Chris Birkbeck. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chris Birkbeck
+-/
+
+import Mathlib.Data.Int.Interval
+import Mathlib.Tactic.Ring.RingNF
+import Mathlib.Tactic.Linarith
+import Mathlib.Tactic.NormNum.Ineq
+
+/-! # Decomposing `ℤ × ℤ` into squares
+
+We partition `ℤ × ℤ` into squares of the form `Icc (-n) n × Icc (-n) n` for `n : ℕ`. This is useful
+for bounding Eisenstein series.
+-/
+
+open scoped BigOperators Classical
+
+noncomputable section
+
+namespace EisensteinSeries
```

Comment:
Even if the results are useful for Eisenstein series, in principle the file is unrelated.
```suggestion
```